storage/innobase/dict/dict0dd.cc

```diff
@@ -6224,17 +6224,19 @@ static bool dd_update_tablespace_dd_flags(
 
 bool dd_set_encryption_flag(THD *thd, const char *space_name,
                             volatile bool *is_space_being_removed) {
-  auto update_func = [](uint32_t &dd_space_flags) {
+  auto update_func = [](uint32_t &dd_space_flags, dd::Tablespace *dd_space) {
     dd_space_flags |= (1U << FSP_FLAGS_POS_ENCRYPTION);
+    dd_space->options().set("encryption", "Y");
```

Comment:
why do we update DD now? I_S queries use only the in-memory object(fil_space_t) right?

Reply:
This maybe should not be part of this PR. It's been a while, but if I remember correctly I just noticed that encryption threads only update the flags and they should also update the options.